pkg/kube/volume/volume.go

```diff
@@ -183,6 +167,41 @@ func CreatePVCFromSnapshot(ctx context.Context, args *CreatePVCFromSnapshotArgs)
 	return pvc.Name, err
 }
 
+func getPVCRestoreSize(ctx context.Context, args *CreatePVCFromSnapshotArgs) (*resource.Quantity, error) {
+	quantities := []*resource.Quantity{}
+
+	sns, err := snapshot.NewSnapshotter(args.KubeCli, args.DynCli)
+	if err != nil {
+		return nil, errors.Wrap(err, "Failed to get snapshotter")
+	}
+	snap, err := sns.Get(ctx, args.SnapshotName, args.Namespace)
+	if err != nil {
+		return nil, errors.Wrap(err, "Failed to get snapshot")
+	}
+	if snap.Status != nil && snap.Status.RestoreSize != nil {
+		quantities = append(quantities, snap.Status.RestoreSize)
+	}
+	if args.RestoreSize != "" {
+		s, err := resource.ParseQuantity(args.RestoreSize)
+		if err != nil {
+			return nil, fmt.Errorf("Failed to parse quantity (%s)", args.RestoreSize)
+		}
```

Comment:
Why fail here if there is a possibility that we have additional values in `quantities`?